sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -232,7 +235,7 @@ protected void resolveAllCallsFromSourceMethod(
         .map(Stmt::asInvokableStmt)
         .forEach(
             stmt ->
-                resolveCall(sourceMethod, stmt)
+                 (boundFunction == null || Boolean.TRUE.equals(boundFunction.apply(sourceMethod, stmt)) ? resolveCall(sourceMethod, stmt) : Stream.<MethodSignature> empty())
```

Comment:
1) instead of null lets make use of a method that just returns always true
2) lets use BiPredicate as type to reduce unnecessary AutoBoxing and simplify the condition check  to just the method call

Reply:
will make adjustment as suggested